tools/generate_tables_sql_and_testNames.php

```diff
@@ -1,118 +1,224 @@
+                ."*/"
+            );
+        }
+        fclose($fp);
     }
-    $output.="PRIMARY KEY  (`CommentID`)\n) ENGINE=InnoDB DEFAULT CHARSET=utf8;\n";
-    print "Filename: $filename\n";
-    $fp=fopen($filename, "w");
-    fwrite($fp, $output);
-    fwrite($fp, "REPLACE INTO test_names (Test_name, Full_name, Sub_group) VALUES ('$tablename', '$title', 1);\n");
-    foreach($pages as $pageNo => $page) {
-        fwrite($fp, "INSERT INTO instrument_subtests (Test_name, Subtest_name, Description) VALUES ('$tablename', '" . $tablename  . "_page" . ($pageNo+1) . "', '$page');\n");
-
+    // Execute the SQL tasks if no error_message.
+    if ($db->isConnected() && empty($error_message)) {
+        if (!$db->tableExists($sql_query_table_name)) {
+            // Table doesn't exist!
+            $sql_query_statement .= implode($sql_query_columns);
+            $sql_query_statement .=
+                "PRIMARY KEY  (`CommentID`)\n) ENGINE=InnoDB "
+                ."DEFAULT CHARSET=utf8;\n";
+            $statement            = $db->prepare($sql_query_statement);
+            $statement->execute();
+        } else {
+            $error_message = "Instrument(s) file uses existing name.";
+        }
+        $statement = $db->prepare(
+            'REPLACE INTO test_names (Test_name, Full_name, Sub_group)'
+            .' VALUES (?, ?, 1)'
+        );
+        $statement->bindParam(1, $sql_query_table_name, PDO::PARAM_STR);
+        $statement->bindParam(2, $title, PDO::PARAM_STR);
+        if ($execute_mysql_query) {
+            $statement->execute();
+        }
+        foreach ($pages as $page_number => $page) {
+            $statement    = $db->prepare(
+                'INSERT INTO instrument_subtests'
+                .' (Test_name, Subtest_name, Description)'
+                .' VALUES (?, ?, ?)'
+            );
+            $subtest_name = $sql_query_table_name . "_page" . ($page_number+1);
+            $statement->bindParam(1, $sql_query_table_name, PDO::PARAM_STR);
+            $statement->bindParam(2, $subtest_name, PDO::PARAM_STR);
+            $statement->bindParam(3, $page, PDO::PARAM_STR);
+            if ($execute_mysql_query) {
+                $statement->execute();
+            }
+        }
     }
-    fclose($fp);
 }
-
-echo "\n\Table SQL Generation complete\n";
-
-function enumizeOptions($options, $table, $name){
-    $options=explode("{-}",$options);
-    foreach($options as $option){
-        $option=explode("=>",$option);
-        if($option[0]!='NULL'){
-            $enum[]=$option[0];
+if ($error_message) {
+    echo $error_message;
+}
+/**
+ * Creates a partial MySQL enum statement based on options extracted from
+ * a LINST file (which is in turn based on an HTML select field).
+ *
+ * @param array  $options the enum values.
+ * @param string $table   the table name used.
+ * @param string $name    the enum name used.
+ *
+ * @return string
+ */
+function enumizeOptions($options, $table, $name)
```

Comment:
The $table and $name parameter are not used anymore. I would remove them from the function def. (and adjust call to this function

Reply:
Not use in this function (enumizeOptions).